### ISSUE TITLE
XMLRPC TestPlan.create does not create complete/valid TestPlans

### ISSUE THREAD

**Issue by tash** [Nitrate/Nitrate#500]
When a TestPlan is created via XMLRPC API and an user tries to edit it through ``/plan/<plan_id>/edit/``, an error is produced. It appears that on creation through XMLRPC tcms.TestPlan.create, no owner is filled. Even updates on the new testplan adding an owner through XMLRPC tcms.TestPlan.update do not fix the RelatedObjectDoesNotExist Exception.
**Observer Behaviour**
- the Tesplan is created as Model object.
- an Exception is displayed in the browser
```
Environment:


Request Method: GET
Request URL: https://127.0.0.1/plan/27/edit/

Django Version: 2.2.8
Python Version: 3.6.9
Installed Applications:
('django.contrib.admin',
 'django.contrib.admindocs',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.messages',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.staticfiles',
 'django_comments',
 'kobo.django.xmlrpc',
 'tinymce',
 'tcms.auth.apps.AppConfig',
 'tcms.comments.apps.AppConfig',
 'tcms.linkreference',
 'tcms.logs',
 'tcms.issuetracker',
 'tcms.integration.errata',
 'tcms.management',
 'tcms.profiles',
 'tcms.testcases',
 'tcms.testplans',
 'tcms.testruns',
 'tcms.xmlrpc.apps.AppConfig',
 'tcms.report',
 'tcms.core.apps.AppConfig')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'tcms.core.middleware.CsrfDisableMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware')

Traceback:

File "/usr/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/lib/python3.6/site-packages/django/views/decorators/http.py" in inner
  40.             return func(request, *args, **kwargs)

File "/usr/lib/python3.6/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)

File "./tcms/testplans/views.py" in edit
  622.             'auto_to_plan_owner': tp.email_settings.auto_to_plan_owner,

File "/usr/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py" in __get__
  415.                     self.related.get_accessor_name()

Exception Type: RelatedObjectDoesNotExist at /plan/27/edit/
Exception Value: TestPlan has no email_settings.
```

**Expected Behaviour**
The edit view for the given plan_id is displayed.

**Steps of recreation**:
Precondition: DEBUG Flag is set to ``True`` in tcms.settings.
1. Create a TestPlan via XMLRPC API

Example:
```python
import xmlrpc.client

with xmlrpc.client.ServerProxy(url, verbose=False, allow_none=True) as proxy:
    values = {
            'product': "1",
            'name': 'TestPlan Python Skript',
            'type': plan_type_segment["id"],
            'product_version': "1",
            'default_product_version': "1",
            'text': "This is a Python script test",
            'is_active': True,
            }
    multicall = xmlrpc.client.MultiCall(proxy)
    multicall.Auth.login({'username': "valid_user", 'password': "valid_password"})
    multicall.TestPlan.create(values)
    session = multicall()[0]
    testplan = multicall()[1]
    print(testplan)
```
2. Navigate to <host-addr>/plan/<plan_id>/edit/

**Comment by tash** [comment]
The following method could use some EmailSetting (ie. ``TestPlanEmailSettings.objects.create(plan=tp)``):
https://github.com/Nitrate/Nitrate/blob/2c3c60826730687ccb3e9347bf15798074926dd7/src/tcms/xmlrpc/api/testplan.py#L136-L201

**Comment by tash** [comment]
Although, Email Settings should be created implicitly on every TestPlan  and thus be solved in src/tcms/testplans/models.py ?

**Comment by tkdchen** [comment]
Yeah, every TestPlan must have an email settings object created implicitly. TestCase has such a same case as well and the related email settings object creation happens on access to property `TestCase.emailing`.

But, for now, I think it should be a good idea to create a related email settings object in a TestPlan `post_save` signal handler. Once it is done, the same solution could be implemented for TestCase.

**Comment by tkdchen** [comment]
Fixed in f366432c1a33b472679335a9bec828375e2b9e7a
